pkg/cmd/operator/gatherbase.go

```diff
@@ -0,0 +1,152 @@
+package operator
+
+import (
+	"flag"
+	"fmt"
+	"math"
+	"os"
+	"path/filepath"
+
+	"github.com/scylladb/scylla-operator/pkg/gather/collect"
+	"github.com/scylladb/scylla-operator/pkg/genericclioptions"
+	"github.com/scylladb/scylla-operator/pkg/version"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	apierrors "k8s.io/apimachinery/pkg/util/errors"
+	utilrand "k8s.io/apimachinery/pkg/util/rand"
+	kgenericclioptions "k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/discovery"
+	cacheddiscovery "k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	cliflag "k8s.io/component-base/cli/flag"
+	"k8s.io/klog/v2"
+)
+
+type GatherBaseOptions struct {
+	gathererName string
+	configFlags  *kgenericclioptions.ConfigFlags
+
+	kubeClient      kubernetes.Interface
+	dynamicClient   dynamic.Interface
+	discoveryClient discovery.DiscoveryInterface
+
+	DestDir              string
+	CollectManagedFields bool
+	LogsLimitBytes       int
+	KeepGoing            bool
+}
+
+func NewGatherBaseOptions(gathererName string, keepGoing bool) *GatherBaseOptions {
+	return &GatherBaseOptions{
+		gathererName: gathererName,
+		configFlags: kgenericclioptions.NewConfigFlags(true).WithWrapConfigFn(func(c *rest.Config) *rest.Config {
+			c.UserAgent = genericclioptions.MakeVersionedUserAgent(fmt.Sprintf("scylla-operator-%s", gathererName))
+			// Don't slow down artificially.
+			c.QPS = math.MaxFloat32
+			c.Burst = math.MaxInt
+			return c
+		}),
+		DestDir:              "",
+		CollectManagedFields: false,
+		LogsLimitBytes:       0,
+		KeepGoing:            keepGoing,
+	}
+}
+
+func (o *GatherBaseOptions) AddFlags(flagset *pflag.FlagSet) {
+	o.configFlags.AddFlags(flagset)
+
+	flagset.StringVarP(&o.DestDir, "dest-dir", "", o.DestDir, "Destination directory where to store the artifacts.")
+	flagset.IntVarP(&o.LogsLimitBytes, "log-limit-bytes", "", o.LogsLimitBytes, "Maximum number of bytes collected for each log file, 0 means unlimited.")
+	flagset.BoolVarP(&o.CollectManagedFields, "managed-fields", "", o.CollectManagedFields, "Controls whether metadata.managedFields should be collected in the resource dumps.")
+	flagset.BoolVarP(&o.KeepGoing, "false", "", o.KeepGoing, "Controls whether the collection should proceed to other resources over collection errors, accumulating errors.")
```

Comment:
fixing the name, thx

I like `keep-going` more because the default is to fail, contrary to `fail-fast` where the expectation is by default to run as far as it can. Also this is modeled after ` make --keep-going`.